src/main/java/net/neoforged/neoforge/event/entity/player/PlayerEvent.java

```diff
@@ -387,39 +387,130 @@ public String getPlayerUUID() {
         }
     }
 
+    /**
+     * An event triggered when a player crafts an item.
+     * <p>
+     * This event is fired at the following stages:
+     * <ul>
+     * <li>After clicking the craft button in the crafting interface.</li>
+     * <li>Before awarding the used recipe and resizing the stacks.</li>
+     * </ul>
+     * <p>
+     * The event is fired via {@link EventHooks#firePlayerCraftingEvent(Player, ItemStack, Container)}<br>
+     * and is posted to the {@link NeoForge#EVENT_BUS}.
+     */
     public static class ItemCraftedEvent extends PlayerEvent {
-        private final ItemStack crafting;
+        private final ItemStack result;
         private final Container craftMatrix;
 
-        public ItemCraftedEvent(Player player, ItemStack crafting, Container craftMatrix) {
+        public ItemCraftedEvent(Player player, ItemStack result, Container craftMatrix) {
             super(player);
-            this.crafting = crafting;
+            this.result = result;
             this.craftMatrix = craftMatrix;
         }
 
+        /**
+         * {@return the item that was crafted (ex. Diamond sword)}
+         */
         public ItemStack getCrafting() {
```

Comment:
This Method name, and the same method from the `ItemSmeltedEvent` are misleading.

I recommend renaming them to `getResult` or `getCrafted`(for the `ItemSmeltedEvent`, `getSmelted`) since they return the result, and not the matrix/item that is being smelted.


Since this is a Breaking Change i was requested in the Discord to make a comment about it